hail_scripts/v02/utils/hail_utils.py

```diff
@@ -0,0 +1,123 @@
+import hail as hl
+import logging
+
+from hail_scripts.v02.utils.computed_fields.variant_id import get_expr_for_orig_alt_alleles
+
+logger = logging.getLogger()
+
+
+def import_table(
+        table_path: str,
+        min_partitions: int = None,
+        impute: bool = False,
+        no_header: bool = False,
+        force_bgz: bool = True,
+        comment: str = "#",
+        types: dict = None,
+    ):
+
+    logger.info(f"\n==> import table: {table_path}")
+
+    ht = hl.import_table(
+        table_path,
+        impute=impute,
+        no_header=no_header,
+        force_bgz=force_bgz,
+        comment=comment,
+        types=types,
+        min_partitions=min_partitions)
+
+    ht = ht.annotate_globals(sourceFilePath=table_path)
+
+    return ht
+
+
+def import_vcf(
+        vcf_path: str,
+        genome_version: str,
+        min_partitions: int = None,
+        force_bgz: bool = True,
+        drop_samples: bool = False,
+        skip_invalid_loci: bool = False):
+    """Import vcf and return MatrixTable.
+
+    :param str vcf_path: MT to annotate with VEP
+    :param str genome_version: "37" or "38"
+    :param int min_partitions: min partitions
+    :param bool force_bgz: read .gz as a bgzipped file
+    :param bool drop_samples: if True, discard genotype info
+    :param bool skip_invalid_loci: if True, skip loci that are not consistent with the reference_genome.
+    """
+
+    if genome_version not in ("37", "38"):
+        raise ValueError(f"Invalid genome_version: {genome_version}")
+
+    logger.info(f"\n==> import vcf: {vcf_path}")
+
+    # add (or remove) "chr" prefix from vcf chroms so they match the reference
+    ref = hl.get_reference(f"GRCh{genome_version}")
+    contig_recoding = {
+        **{ref_contig.replace("chr", ""): ref_contig for ref_contig in ref.contigs if "chr" in ref_contig},
+        **{f"chr{ref_contig}": ref_contig for ref_contig in ref.contigs if "chr" not in ref_contig}}
+
+    mt = hl.import_vcf(
+        vcf_path,
+        reference_genome=f"GRCh{genome_version}",
+        contig_recoding=contig_recoding,
+        min_partitions=min_partitions,
+        force_bgz=force_bgz,
+        drop_samples=drop_samples,
+        skip_invalid_loci=skip_invalid_loci)
+
+    mt = mt.annotate_globals(sourceFilePath=vcf_path, genomeVersion=genome_version)
+
+    # ensure that va.wasSplit and va.aIndex are defined before calling split_multi() since split_multi() doesn't define these if all variants are bi-allelic
+    mt = mt.annotate_rows(was_split=False, a_index=1, original_alt_alleles=get_expr_for_orig_alt_alleles(mt))
+    mt = hl.split_multi_hts(mt)
+    mt = mt.key_rows_by(**hl.min_rep(mt.locus, mt.alleles))
+
+    return mt
+
+
+def read_mt(mt_path: str):
```

Comment:
To me, these functions that wrap one Hail function obfuscate the code. Someone familiar with Hail knows what `hl.read_matrix_table` does, but they likely have to look up what `hail_scripts/v02/utils/hail_utils.read_mt` does.

Reply:
I added the import_vcf wrapper to do the several things (like split_multi) that I always do after importing, and then added the other ones for completeness. I see your point though and am fine with dropping these.